scripts/get-hypper

```diff
@@ -0,0 +1,263 @@
+# the package manager for Go: https://github.com/Masterminds/glide.sh/blob/master/get
+
+: "${BINARY_NAME:=hypper}"
+: "${USE_SUDO:=true}"
+: "${DEBUG:=false}"
+: "${VERIFY_CHECKSUM:=true}"
+: "${HYPPER_INSTALL_DIR:=/usr/local/bin}"
+
+
+HAS_CURL="$(type "curl" &> /dev/null && echo true || echo false)"
+HAS_WGET="$(type "wget" &> /dev/null && echo true || echo false)"
+HAS_OPENSSL="$(type "openssl" &> /dev/null && echo true || echo false)"
+
+# initArch discovers the architecture for this system.
+initArch() {
+  ARCH=$(uname -m)
+  case $ARCH in
+    armv5*) ARCH="armv5";;
+    armv6*) ARCH="armv6";;
+    armv7*) ARCH="arm";;
+    aarch64) ARCH="arm64";;
+    x86) ARCH="386";;
+    x86_64) ARCH="x86_64";;
+    i686) ARCH="386";;
+    i386) ARCH="386";;
+  esac
+}
+
+# initOS discovers the operating system for this system.
+initOS() {
+  OS=$(uname | tr '[:upper:]' '[:lower:]')
+
+  case "$OS" in
+    # Minimalist GNU for Windows
+    mingw*) OS='windows';;
+  esac
+}
+
+# runs the given command as root (detects if we are root already)
+runAsRoot() {
+  if [ $EUID -ne 0 ] && [ "$USE_SUDO" = "true" ]; then
+    sudo "${@}"
+  else
+    "${@}"
+  fi
+}
+
+# verifySupported checks that the os/arch combination is supported for
+# binary builds, as well whether or not necessary tools are present.
+verifySupported() {
+  local supported="darwin-x86_64\ndarwin-arm64\nlinux-386\nlinux-x86_64\nlinux-arm\nlinux-arm64\nlinux-ppc64le\nlinux-s390x\nwindows-x86_64"
+  if ! echo "${supported}" | grep -q "${OS}-${ARCH}"; then
+    echo "No prebuilt binary for ${OS}-${ARCH}."
+    echo "To build from source, go to https://github.com/itxaka/hypper"
+    exit 1
+  fi
+
+  if [ "${HAS_CURL}" != "true" ] && [ "${HAS_WGET}" != "true" ]; then
+    echo "Either curl or wget is required"
+    exit 1
+  fi
+
+  if [ "${VERIFY_CHECKSUM}" == "true" ] && [ "${HAS_OPENSSL}" != "true" ]; then
+    echo "In order to verify checksum, openssl must first be installed."
+    echo "Please install openssl or set VERIFY_CHECKSUM=false in your environment."
+    exit 1
+  fi
+}
+
+# checkDesiredVersion checks if the desired version is available.
+checkDesiredVersion() {
+  if [ "x$DESIRED_VERSION" == "x" ]; then
+    # Get tag from release URL
+    local latest_release_url="https://api.github.com/repos/itxaka/hypper/releases/latest"
+    if [ "${HAS_CURL}" == "true" ]; then
+      TAG=$(curl -Ls $latest_release_url | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/')
+    elif [ "${HAS_WGET}" == "true" ]; then
+      TAG=$(wget $latest_release_url -O - 2>&1 | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/')
+    fi
+  else
+    TAG=$DESIRED_VERSION
+  fi
+}
+
+# checkHypperInstalledVersion checks which version of hypper is installed and
+# if it needs to be changed.
+checkHypperInstalledVersion() {
+  if [[ -f "${HYPPER_INSTALL_DIR}/${BINARY_NAME}" ]]; then
+    local version
+    version=$("${HYPPER_INSTALL_DIR}/${BINARY_NAME}" version --template="{{ .Version }}")
```

Comment:
Should we implement a `hypper version` command before merging this?

Reply:
For now it doesnt matter as we dont have any released versions that we can compare, so this will always update to lates, but its  a good idea to have that, plus Im sure its a very simple command.

Reply:
this is already done!